### PR TITLE
Address DNS-SD issues

### PR DESCRIPTION
**Pull request by chrisdecenzo**
#### Problem
* Issue #16237: DNS-SD pairing hint field is incorrect when using BCM and already commissioned
* Issue #16624: Extended Discovery cannot be set to 24 hours 
* Issue #16680: tv-casting broken when tv-app has IPv4 disabled

#### Change overview
* use "initial" pairing hint value when not yet commissioned, and not only during ECM (current logic)
* change extended discovery timeout value from int16 to int32 to allow longer values
* do not prefer v4 address in UDC when v4 is disabled (prefer v6 address)

#### Testing
How was this tested? (at least one bullet point required)
* Run sample device, use chip-tool to discover commissionables and check the PH values when not commissioned, and when put in commissioning mode via BCM and ECM.

### CODE REVIEW

File: src/include/platform/CHIPDeviceConfig.h [hunk 1]
```diff
@@ -1237,7 +1237,7 @@
  */
 #define CHIP_DEVICE_CONFIG_DISCOVERY_DISABLED 0
 #define CHIP_DEVICE_CONFIG_DISCOVERY_NO_TIMEOUT -1
-#define CHIP_DEVICE_CONFIG_EXTENDED_DISCOVERY_TIMEOUT_SECS (15 * 60)
+#define CHIP_DEVICE_CONFIG_EXTENDED_DISCOVERY_TIMEOUT_SECS (24 * 60 * 60)
```
Comment (Damian-Nordic) [inline src/include/platform/CHIPDeviceConfig.h:1240]: Why is this change needed? I think the spec recommends limiting announcements by default to up to 15 min.

Comment (chrisdecenzo) [reply src/include/platform/CHIPDeviceConfig.h:1240]: Can you share where you see that? My understanding is that the 15 minute recommendation is for commissioning mode and not extended discovery.

Comment (Damian-Nordic) [reply src/include/platform/CHIPDeviceConfig.h:1240]: > The default behavior for Commissionable Node Discovery SHOULD default to limiting announcement as
defined in Section 5.4.2.3, “Announcement Duration”
It doesn't specifically say anything about the commissioning mode and the text is part of the "Extended discovery" section so my understanding is that the recommendation applies to the extended discovery as well.

Comment (chrisdecenzo) [reply src/include/platform/CHIPDeviceConfig.h]: ok, I'm not sure on that sentence either but happy to keep the default at 15 mins and so I changed this back and added an override for the tv-app.

**Review by Damian-Nordic — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]
